rpcs3/Emu/RSX/Common/TextureUtils.cpp

```diff
@@ -66,7 +66,7 @@ struct copy_unmodified_block
 		{
 			// Fast copy
 			const auto data_length = src_pitch_in_block * words_per_block * row_count * depth;
-			copy(dst, src.subspan(0, data_length));
+			memcpy(dst.data(), src.data(), data_length * sizeof(T));
```

Comment:
This is wrong when T != U. I already tried using memcpy when T == U but all compilers were giving the same code, just using memmove instead of memcpy and performance was the same. Usually the gpu decoder will handle format conversion stuff though.

Reply:
Do you mean this function shouldn’t get called at all, and something isn’t working on the Vulkan side?

I indeed missed the T ≠ U case, but given the huge difference in the profile (with both clang++ and g++) it seems very valid to have this optimisation, also elsewhere as long as T = U.

Reply:
I checked the decompiled code, it was just a call to memmove, so I intentionally reversed the optimization since it had no benefit. Now there are cases where this function is called, rightly so, and likely with T != U. This is usually for smaller textures smaller than can properly utilize the GPU. If clang and gcc are not generating the proper assembly that is another issue altogether.

Reply:
Wait, my profiling was actually done on top of your #6941 branch, hence the widely different profile compared to before, with deswizzling done on the GPU and then you doing a CPU-side copy, which is what I saw here and optimised.

The overwhelmingly common case was T=u32 and U=u32 in this scene.

Reply:
So I’ve made a testcase here: https://godbolt.org/z/FtnK52 (try it with MSVC, gcc and clang).

Only MSVC generates code that calls to `memmove()`, all other compilers will go through the elements one by one in the `gsl::span` and increment the counter.

Interestingly, the same code but using `std::vector` leads to proper code on all of the compilers, seemingly pointing to a bug in Microsoft’s span implementation.